core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/terminologyregistry/core/request/CodeSystemRequests.java

```diff
@@ -47,5 +57,46 @@ public static CodeSystemVersionSearchRequestBuilder prepareSearchCodeSystemVersi
 	public static CodeSystemVersionCreateRequestBuilder prepareNewCodeSystemVersion() {
 		return new CodeSystemVersionCreateRequestBuilder();
 	}
+
+	/**
+	 * Returns all {@link CodeSystemEntry}s from all repositories.
+	 * @param context
+	 * @return
+	 */
+	public static List<CodeSystemEntry> getAllCodeSystems(ServiceProvider context) {
+		final Repositories repositories = RepositoryRequests.prepareSearch()
+			.all()
+			.build()
+			.execute(context);
+		
+		return repositories.getItems()
+			.stream()
+			.filter( repository -> RepositoryInfo.Health.GREEN == repository.health() )
```

Comment:
What is the general consensus around repositories with `YELLOW` health? Can those still be read?

Reply:
Yes, they can be read, I'll add yellow health state as well here.